src/uwtools/tests/test_cli.py

```diff
@@ -234,62 +352,26 @@ def test_main_raises_exception(capsys):
 
 
 def test__parse_args():
-    raw_args = ["foo", "--bar", "88"]
+    raw_args = ["testing", "--bar", "88"]
     with patch.object(cli, "Parser") as Parser:
         cli._parse_args(raw_args)
         Parser.assert_called_once()
         parser = Parser()
         parser.parse_args.assert_called_with(raw_args)
 
 
-@pytest.mark.parametrize(
-    "vals",
-    [
-        (ns(file_1_path=None, file_1_format=None), "--file-1-path", "--file-1-format"),
-        (ns(file_2_path=None, file_2_format=None), "--file-2-path", "--file-2-format"),
-        (ns(input_file=None, input_format=None), "--input-file", "--input-format"),
-        (ns(output_file=None, output_format=None), "--output-file", "--output-format"),
-        (ns(values_file=None, values_format=None), "--values-file", "--values-format"),
-    ],
-)
-def test__set_formats_fail(capsys, vals):
-    # When reading/writing from/to stdin/stdout, the data format must be specified, since there is
-    # no filename to deduce it from.
-    args, path_arg, fmt_arg = vals
-    with raises(SystemExit):
-        cli._set_formats(args)
-    assert f"Specify {fmt_arg} when {path_arg} is not specified" in capsys.readouterr().err
-
-
-def test__set_formats_pass_explicit():
-    # Accept explcitly-specified format, whatever it is.
-    args = ns(input_file="/path/to/input.txt", input_format="jpg")
-    args = cli._set_formats(args)
-    assert args.input_format == "jpg"
-
-
-@pytest.mark.parametrize("fmt", vars(FORMAT).keys())
-def test__set_formats_pass_implicit(fmt):
-    # The format is correctly deduced for a file with a known extension.
-    args = ns(input_file=f"/path/to/input.{fmt}", input_format=None)
-    args = cli._set_formats(args)
-    assert args.input_format == vars(FORMAT)[fmt]
+# Helper functions
 
 
-# Helper functions
+def submodes(parser: Parser) -> List[str]:
+    # Return submodes (named subparsers) belonging to the given parser. For some background, see
+    # https://stackoverflow.com/questions/43688450.
+    if actions := [x for x in parser._actions if isinstance(x, _SubParsersAction)]:
+        return list(actions[0].choices.keys())
+    return []
```

Comment:
I appreciate this change. I _think_ I'm now reading code that tells me in itself that it's looking through the parser for the first instance of a subparser action and returning that action's choices.